python/pyarrow/_fs.pyx

```diff
@@ -100,6 +100,53 @@ cdef class FileInfo(_Weakrefable):
         If given, the filesystem entry size in bytes.  This should only
         be given if `type` is `FileType.File`.
 
+    Examples
+    --------
+    Generate a file:
+
+    >>> from pyarrow import fs
+    >>> local = fs.LocalFileSystem()
+    >>> with local.open_output_stream('/tmp/fileinfo.dat') as stream:
+    ...     stream.write(b'data')
+    ...
+    4
+
+    Get FileInfo object using `get_file_info()`:
+
+    >>> local.get_file_info('/tmp/fileinfo.dat')
+    <FileInfo for '/tmp/fileinfo.dat': type=FileType.File, size=4>
+
+    or FileInfo constructor:
+
+    >>> fs.FileInfo('/tmp/fileinfo.dat', type=fs.FileType.File)
+    <FileInfo for '/tmp/fileinfo.dat': type=FileType.File, size=None>
```

Comment:
I am not fully sure we should show the constructor (I don't know of a reason that someone would want to create one manually?)

Reply:
Not sure either =) Will remove it.